experimental/segmenter/src/lstm.rs

```diff
@@ -4,21 +4,101 @@
 
 use icu_segmenter_lstm::lstm::Lstm;
 use std::char::decode_utf16;
+use std::str::Chars;
 
 // TODO:
 // json file is big, So I should use anoher binary format like npy.
 // But provided npy uses tensorflow dtype.
 const THAI_MODEL: &[u8; 373466] = include_bytes!(
     "../../segmenter_lstm/tests/testdata/Thai_codepoints_exclusive_model4_heavy/weights.json"
 );
+const BURMESE_MODEL: &[u8; 475209] =
+    include_bytes!("../data/Burmese_codepoints_exclusive_model4_heavy/weights.json");
 
 lazy_static! {
     static ref THAI_LSTM: Lstm = {
         let lstm_data = serde_json::from_slice(THAI_MODEL).expect("JSON syntax error");
         Lstm::try_new(lstm_data).unwrap()
     };
+    static ref BURMESE_LSTM: Lstm = {
+        let lstm_data = serde_json::from_slice(BURMESE_MODEL).expect("JSON syntax error");
+        Lstm::try_new(lstm_data).unwrap()
+    };
+}
+
+// LSTM model depends on language, So we have to switch models per language.
+
+#[derive(PartialEq)]
+enum Language {
+    BURMESE,
+    THAI,
+    UNKNOWN,
+}
+
+fn get_language(codepoint: u32) -> Language {
+    match codepoint {
+        0xe01..=0xe7f => Language::THAI,
+        0x1000..=0x109f => Language::BURMESE,
+        _ => Language::UNKNOWN,
+    }
+}
+
+fn get_best_lstm_model(codepoint: u32) -> &'static Lstm {
+    let lang = get_language(codepoint);
+    match lang {
+        Language::THAI => &*THAI_LSTM,
+        Language::BURMESE => &*BURMESE_LSTM,
+        _ => panic!("Unsupported"),
+    }
+}
+
+struct LanguageIterator<'a> {
+    input: Chars<'a>,
+    last: Option<char>,
+}
+
+impl<'a> LanguageIterator<'a> {
+    pub fn new(input: &'a str) -> Self {
+        Self {
+            input: input.chars(),
+            last: None,
+        }
+    }
+}
+
+impl<'a> Iterator for LanguageIterator<'a> {
```

Comment:
Suggestion: Could you add a document for this iterator? It seems to iterate on substring of the same language listed in `enum Language`.